 Update bottle to 0.12.11

Fixes: #670

Currently ycmd is only working with bottle that comes with ycmd.
It is really an old version of bottle and ycmd is not compatible with
recent version.

A Debian package must use system-dependencies instead of third_party modules.

This patch is updating bottle and making ycmd to compatible with latest
bottle.

url: https://github.com/onur/ycmd/commit/d323691f59c90904788b14b1d9c0b76e5f4ac542#diff-9152b48ca0c1cfe237a8ec11461ab367L25
author: Onur Aslan (onur)
----
diff -urp ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/bottle_utils.py ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/bottle_utils.py
--- ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1.orig/ycmd/bottle_utils.py	2017-01-08 10:06:43.000000000 -0800
+++ ycmd-c9a47452dafaf81b65db93f77d0fe23c7f539aa1/ycmd/bottle_utils.py	2017-01-15 23:48:12.954819630 -0800
@@ -24,7 +24,7 @@ standard_library.install_aliases()
 from builtins import *  # noqa
 
 from future.utils import PY2
-from ycmd.utils import ToBytes, ToUnicode
+from ycmd.utils import ToUnicode
 import bottle
 
 
@@ -39,5 +39,6 @@ import bottle
 # making life easier for codebases that work across versions, thus preventing
 # tracebacks in the depths of WSGI server frameworks.
 def SetResponseHeader( name, value ):
-  name = ToBytes( name ) if PY2 else ToUnicode( name )
-  bottle.response.set_header( name, ToUnicode( value ) )
+  name = name.encode( 'utf8' ) if PY2 else ToUnicode( name )
+  value = value.encode( 'utf8' ) if PY2 else ToUnicode( value )
+  bottle.response.set_header( name, value )
